test/puppetlabs/dujour/version_check_test.clj

```diff
@@ -19,14 +25,26 @@
            (get-coords {:group-id "foo.foo"
                         :artifact-id "foo"})))))
 
+(defn parse-params [params encoding]
+  (let [params (codec/form-decode params (str encoding))]
+    (if (map? params) params {})))
+
 (defn update-available-app
-  [_]
-  {:status 200
-   :body (json/generate-string {:newer true
-                                :link "http://foo.com"
-                                :message "Howdy!"
-                                :product "foo"
-                                :version "9000.0.0"})})
+  [req]
+  (let [params (parse-params (:query-string req) "UTF-8")
+        host-id (params "host-id")
+        site-id (params "site-id")
+        certname (params "certname")
+        cacert (params "cacert")]
+    {:status 200
+     :body (json/generate-string {:newer true
+                                  :link "http://foo.com"
+                                  :message (or (json/generate-string {:host-id host-id
```

Comment:
I don't think this `json/generate-string` is required since we're already calling it again outside of the map a few lines up, right? Also what is the `or` here for?